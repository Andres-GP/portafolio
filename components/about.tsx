import { Card } from "@/components/ui/card";

export function About() {
  const skills = [
    "Next.js",
    "React",
    "Redux",
    "HTML5",
    "CSS3",
    "Javascript",
    "Typescript",
    "Sass",
    "Material UI",
    "Tailwind",
    "Ant Design",
    "Bootstrap",
    "Semmantic UI",
    "Element UI",
    "Cypress",
    "Jest",
    "React Testing Library",
    "Nodejs",
    "Express.js",
    "AWS",
    "Azure",
    "Git",
    "Gitlab",
    "Github",
    "Docker",
    "Adobe",
    "Figma",
  ];

  return (
    <section id="about" className="py-24">
      <div className="space-y-12">
        <div className="space-y-2">
          <h2 className="text-3xl font-bold tracking-tight text-foreground sm:text-4xl">
            <span className="text-primary font-mono text-xl">01.</span> About Me
          </h2>
          <div className="h-px w-64 bg-border" />
        </div>
        <div className="grid gap-12 lg:grid-cols-2">
          <div className="space-y-4">
            <p className="text-muted-foreground leading-relaxed">
              Hello! I'm Andrés, a passionate developer who loves bringing ideas
              to life through clean, elegant code and thoughtful design. My
              journey into web development began many years ago when I decided
              to experiment with creating my very first website — a small
              project that quickly grew into a deep curiosity for how things
              work on the web. Since then, I've been continuously learning,
              exploring new technologies, and improving my craft one project at
              a time.
            </p>

            <p className="text-muted-foreground leading-relaxed">
              Fast-forward to today, and I’ve had the privilege of collaborating
              with talented teams across different companies and industries. My
              main focus now is on building accessible, inclusive, and
              high-performing digital experiences that not only look good but
              also solve real problems for real people. I’m driven by curiosity,
              precision, and the desire to create products that blend usability,
              performance, and creativity — and I’m always excited about what’s
              coming next in the ever-evolving world of web development.
            </p>
          </div>
          <Card className="p-6 hover:border-primary/50 transition-colors">
            <h3 className="mb-4 text-xl font-semibold text-foreground">
              Technologies I work with:
            </h3>
            <ul className="grid grid-cols-2 gap-2">
              {skills.map((skill) => (
                <li
                  key={skill}
                  className="flex items-center gap-2 text-sm text-muted-foreground"
                >
                  <span className="text-primary">▹</span>
                  {skill}
                </li>
              ))}
            </ul>
          </Card>
        </div>
      </div>
    </section>
  );
}
