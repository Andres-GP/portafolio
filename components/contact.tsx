"use client";

import { useState } from "react";
import { Button } from "@/components/ui/button";
import { Card } from "@/components/ui/card";
import { Mail, MapPin, Phone } from "lucide-react";

export function Contact() {
  const [formData, setFormData] = useState({
    name: "",
    email: "",
    message: "",
  });

  const [status, setStatus] = useState<
    "idle" | "loading" | "success" | "error"
  >("idle");

  const handleSubmit = async (e: React.FormEvent) => {
    e.preventDefault();
    setStatus("loading");

    try {
      const res = await fetch("/api/send", {
        method: "POST",
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify(formData),
      });

      if (res.ok) {
        setStatus("success");
        setFormData({ name: "", email: "", message: "" });
      } else {
        setStatus("error");
      }
    } catch (error) {
      console.error(error);
      setStatus("error");
    } finally {
      setTimeout(() => setStatus("idle"), 4000);
    }
  };

  return (
    <section id="contact" className="py-24">
      <div className="space-y-12">
        <div className="space-y-2">
          <h2 className="text-3xl font-bold tracking-tight text-foreground sm:text-4xl">
            <span className="text-primary font-mono text-xl">04.</span> Get In
            Touch
          </h2>
          <div className="h-px w-64 bg-border" />
        </div>
        <div className="grid gap-8 lg:grid-cols-2">
          <div className="space-y-6">
            <p className="text-lg text-muted-foreground leading-relaxed">
              My inbox is always open. Whether you have an opportunity, a
              question, or just want to say hi, I’ll try my best to get back to
              you as soon as possible!
            </p>
            <div className="space-y-4">
              <Card className="flex items-center gap-4 p-4 hover:border-primary/50 transition-colors">
                <div className="flex h-12 w-12 items-center justify-center rounded-lg bg-primary/10">
                  <Mail className="h-6 w-6 text-primary" />
                </div>
                <div>
                  <p className="text-sm text-muted-foreground">Email</p>
                  <a
                    href="mailto:andresgarciapedreros@gmail.com"
                    className="text-foreground hover:text-primary transition-colors"
                  >
                    andresgarciapedreros@gmail.com
                  </a>
                </div>
              </Card>

              <Card className="flex items-center gap-4 p-4 hover:border-primary/50 transition-colors">
                <div className="flex h-12 w-12 items-center justify-center rounded-lg bg-primary/10">
                  <Phone className="h-6 w-6 text-primary" />
                </div>
                <div>
                  <p className="text-sm text-muted-foreground">Phone</p>
                  <a
                    href="tel:+573058797720"
                    className="text-foreground hover:text-primary transition-colors"
                  >
                    +57 305 879 7720
                  </a>
                </div>
              </Card>

              <Card className="flex items-center gap-4 p-4 hover:border-primary/50 transition-colors">
                <div className="flex h-12 w-12 items-center justify-center rounded-lg bg-primary/10">
                  <MapPin className="h-6 w-6 text-primary" />
                </div>
                <div>
                  <p className="text-sm text-muted-foreground">Location</p>
                  <p className="text-foreground">Bogotá, Colombia</p>
                </div>
              </Card>
            </div>
          </div>

          <Card className="p-8 hover:border-primary/50 transition-colors">
            <form onSubmit={handleSubmit} className="space-y-6">
              <div className="space-y-2">
                <label
                  htmlFor="name"
                  className="text-sm font-medium text-foreground"
                >
                  Name
                </label>
                <input
                  id="name"
                  type="text"
                  value={formData.name}
                  onChange={(e) =>
                    setFormData({ ...formData, name: e.target.value })
                  }
                  className="w-full rounded-lg border border-input bg-background px-4 py-2 text-foreground focus:border-primary focus:outline-none focus:ring-1 focus:ring-primary"
                  placeholder="Your name"
                  required
                />
              </div>

              <div className="space-y-2">
                <label
                  htmlFor="email"
                  className="text-sm font-medium text-foreground"
                >
                  Email
                </label>
                <input
                  id="email"
                  type="email"
                  value={formData.email}
                  onChange={(e) =>
                    setFormData({ ...formData, email: e.target.value })
                  }
                  className="w-full rounded-lg border border-input bg-background px-4 py-2 text-foreground focus:border-primary focus:outline-none focus:ring-1 focus:ring-primary"
                  placeholder="your.email@gmail.com"
                  required
                />
              </div>

              <div className="space-y-2">
                <label
                  htmlFor="message"
                  className="text-sm font-medium text-foreground"
                >
                  Message
                </label>
                <textarea
                  id="message"
                  rows={5}
                  value={formData.message}
                  onChange={(e) =>
                    setFormData({ ...formData, message: e.target.value })
                  }
                  className="w-full rounded-lg border border-input bg-background px-4 py-2 text-foreground focus:border-primary focus:outline-none focus:ring-1 focus:ring-primary resize-none"
                  placeholder="Your message..."
                  required
                />
              </div>

              <Button
                type="submit"
                className="w-full cursor-pointer"
                disabled={status === "loading"}
              >
                {status === "loading" ? "Sending..." : "Send Message"}
              </Button>

              {status === "success" && (
                <p className="text-green-600 text-center text-sm animate-fade-in">
                  ✅ Message sent successfully!
                </p>
              )}
              {status === "error" && (
                <p className="text-red-600 text-center text-sm animate-fade-in">
                  ❌ Something went wrong. Please try again.
                </p>
              )}
            </form>
          </Card>
        </div>
      </div>

      <footer className="mt-24 border-t border-border pt-8 text-center">
        <p className="text-sm text-muted-foreground">
          Built with Next.js & Tailwind CSS. Deployed on Vercel.
        </p>
        <p className="mt-2 text-sm text-muted-foreground">
          © 2025 Andrés García. All rights reserved.
        </p>
      </footer>
    </section>
  );
}
