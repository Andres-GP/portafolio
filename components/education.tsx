import { Card } from "@/components/ui/card";

export function Education() {
  const education = [
    {
      degree: "MicroMasters in Finance (MMS)",
      school: "Massachusetts Institute of Technology (MITx)",
      period: "May 2025 – Dec 2026",
      description: [
        "MITx-certified advanced international program combining rigorous financial theory with hands-on applications in investment, capital markets, and quantitative analysis.",
        "Covered asset valuation, risk management, algorithmic trading, corporate finance, and portfolio optimization techniques.",
        "Focused on strategic decision-making through the use of quantitative models and emerging technologies in global markets.",
        "Engaged in collaborative projects on financial analysis, portfolio management simulations, and asset valuation exercises.",
      ],
      achievements: [
        "Completed advanced coursework in quantitative trading and portfolio optimization",
        "Active participant in finance-related research and simulations",
      ],
    },
    {
      degree: "Certifications",
      school: "Professional Development",
      period: "2020 – 2023",
      description: [
        "Certified in advanced Front-End and Back-End JavaScript Development, including React.js, Vue.js, Angular, and Node.js.",
        "Completed specialized programs in UX/UI Design, Product Design, and Graphic Design, emphasizing usability and product strategy.",
        "Certified in Business Executive and Leadership (Team Management), focused on cross-functional collaboration and agile methodologies.",
        "Developed full-stack and mobile applications with React Native, Firebase, and modern web technologies.",
      ],
      achievements: [
        "Multiple verified certifications in development, design, and leadership.",
        "Recognized for continuous professional growth through technology-driven education.",
      ],
    },
  ];

  return (
    <section id="education" className="py-24">
      <div className="space-y-12">
        <div className="space-y-2">
          <h2 className="text-3xl font-bold tracking-tight text-foreground sm:text-4xl">
            <span className="text-primary font-mono text-xl">03.</span>{" "}
            Education
          </h2>
          <div className="h-px w-64 bg-border" />
        </div>
        <div className="space-y-6">
          {education.map((edu, index) => (
            <Card
              key={index}
              className="p-6 hover:border-primary/50 transition-colors"
            >
              <div className="space-y-4">
                <div className="flex flex-col gap-2 sm:flex-row sm:items-center sm:justify-between">
                  <div>
                    <h3 className="text-xl font-semibold text-foreground">
                      {edu.degree}
                    </h3>
                    <p className="text-primary">{edu.school}</p>
                  </div>
                  <p className="text-sm text-muted-foreground font-mono">
                    {edu.period}
                  </p>
                </div>
                <ul className="space-y-2">
                  {edu.description.map((item, i) => (
                    <li
                      key={i}
                      className="flex gap-2 text-sm text-muted-foreground leading-relaxed"
                    >
                      <span className="text-primary mt-1">▹</span>
                      <span>{item}</span>
                    </li>
                  ))}
                </ul>
                <div className="flex flex-wrap gap-2">
                  {edu.achievements.map((achievement) => (
                    <span
                      key={achievement}
                      className="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
                    >
                      {achievement}
                    </span>
                  ))}
                </div>
              </div>
            </Card>
          ))}
        </div>
      </div>
    </section>
  );
}
