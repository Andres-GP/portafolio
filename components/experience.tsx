import React from "react";
import { Card } from "@/components/ui/card";

export function Experience() {
  const experiences = [
    {
      title: "Frontend Developer",
      company: "Freelance",
      period: "March 2025 – Present · Remote",
      description: [
        "Lead project planning and architecture design for client applications.",
        "Develop and design high-quality web apps with React.js, Next.js, and modern technologies.",
        "Implement scalable micro-frontends and micro-services to ensure flexibility and performance.",
        "Collaborate with clients to deliver intuitive, responsive, and efficient web solutions.",
      ],
      technologies: [
        "React.js",
        "Next.js",
        "AWS",
        "Azure",
        "Node.js",
        "TypeScript",
        "Jest",
        "Express.js",
        "Material-UI",
        "Git",
        "GitHub",
        "GitLab",
        "Figma",
        "HTML5",
        "CSS",
        "Sass",
        "Micro-Frontends",
        "Micro-Services",
      ],
    },
    {
      title: "Web UI Developer (Semi-Senior Advanced)",
      company: "Globant",
      period: "Oct 2023 – March 2025 · Remote",
      description: [
        "Led collaboration with clients through the full development lifecycle — from testing to optimization.",
        "Coached and mentored developers, providing technical feedback and promoting best practices.",
        "Participated in agile project planning and delivery processes to meet product goals.",
        "Drove UI performance improvements and enhanced accessibility in production applications.",
      ],
      technologies: [
        "React.js",
        "Next.js",
        "TypeScript",
        "Node.js",
        "Jest",
        "Material-UI",
        "Azure",
        "AWS",
        "GitLab",
        "Agile",
        "HTML5",
        "CSS",
        "Sass",
        "Micro-Frontends",
        "Micro-Services",
      ],
    },
    {
      title: "Front-End Developer & UX/UI Designer",
      company: "Kovah",
      period: "Dec 2021 – Oct 2023 · Remote",
      description: [
        "Developed and optimized large-scale web applications using React with Redux, Sagas, and Tailwind.",
        "Supervised UX/UI design processes to improve usability and visual consistency using Figma.",
        "Coordinated with team members and project managers for continuous delivery and feature planning.",
        "Led technical interviews to expand and strengthen the development team.",
      ],
      technologies: [
        "React.js",
        "Redux",
        "Sass",
        "Tailwind",
        "Material-UI",
        "Semantic UI",
        "Formik",
        "Yup",
        "Emotion",
        "Framer",
        "Rollup.js",
        "Webpack",
        "HTML5",
        "CSS",
        "TypeScript",
        "JavaScript",
        "Micro-Frontends",
        "Micro-Services",
        "Figma",
      ],
    },
    {
      title: "Front-End Developer",
      company: "A&L Software",
      period: "Jan 2020 – Dec 2021 · Remote",
      description: [
        "Developed, tested, and optimized React-based applications using modern front-end frameworks.",
        "Collaborated with project managers, stakeholders, and clients to deliver high-quality solutions.",
        "Implemented responsive UIs and ensured cross-browser compatibility using modern web standards.",
        "Worked with Next.js, Tailwind, and Ant Design to enhance performance and maintainability.",
      ],
      technologies: [
        "React.js",
        "Next.js",
        "TypeScript",
        "JavaScript",
        "Axios",
        "Ant Design",
        "Tailwind",
        "Bootstrap",
        "Formik",
        "Webpack",
        "HTML5",
        "CSS",
      ],
    },
  ];

  return (
    <section id="experience" className="py-24">
      <div className="space-y-12">
        <div className="space-y-2">
          <h2 className="text-3xl font-bold tracking-tight text-foreground sm:text-4xl">
            <span className="text-primary font-mono text-xl">02.</span>{" "}
            Experience
          </h2>
          <div className="h-px w-64 bg-border" />
        </div>
        <div className="space-y-6">
          {experiences.map((exp, index) => (
            <Card
              key={index}
              className="p-6 hover:border-primary/50 transition-colors"
            >
              <div className="space-y-4">
                <div className="flex flex-col gap-2 sm:flex-row sm:items-center sm:justify-between">
                  <div>
                    <h3 className="text-xl font-semibold text-foreground">
                      {exp.title}
                    </h3>
                    <p className="text-primary">{exp.company}</p>
                  </div>
                  <p className="text-sm text-muted-foreground font-mono">
                    {exp.period}
                  </p>
                </div>
                <ul className="space-y-2">
                  {exp.description.map((item, i) => (
                    <li
                      key={i}
                      className="flex gap-2 text-sm text-muted-foreground leading-relaxed"
                    >
                      <span className="text-primary mt-1">▹</span>
                      <span>{item}</span>
                    </li>
                  ))}
                </ul>
                <div className="flex flex-wrap gap-2">
                  {exp.technologies.map((tech) => (
                    <span
                      key={tech}
                      className="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
                    >
                      {tech}
                    </span>
                  ))}
                </div>
              </div>
            </Card>
          ))}
        </div>
      </div>
    </section>
  );
}
