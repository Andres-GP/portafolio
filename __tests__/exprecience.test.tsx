import React from "react";
import { render, screen } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import { Experience } from "@/components/experience";

vi.mock("@/components/ui/card", () => ({
  Card: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="card">{children}</div>
  ),
}));

describe("<Experience />", () => {
  it("renders all experience cards", () => {
    render(<Experience />);
    const cards = screen.getAllByTestId("card");
    expect(cards.length).toBe(4);
  });

  it("renders each job title and company correctly", () => {
    render(<Experience />);

    expect(screen.getByText("Frontend Developer")).toBeInTheDocument();
    expect(screen.getByText("Freelance")).toBeInTheDocument();

    expect(screen.getByText("Front-End Developer")).toBeInTheDocument();
    expect(screen.getByText("A&L Software")).toBeInTheDocument();
  });

  it("renders periods for each experience", () => {
    render(<Experience />);
    expect(
      screen.getByText(/March 2025 – Present · Remote/i)
    ).toBeInTheDocument();
    expect(
      screen.getByText(/Jan 2020 – Dec 2021 · Remote/i)
    ).toBeInTheDocument();
  });

  it("renders at least one description item per experience", () => {
    render(<Experience />);
    expect(
      screen.getByText(/Lead project planning and architecture design/i)
    ).toBeInTheDocument();

    expect(
      screen.getByText(/Developed and optimized large-scale web applications/i)
    ).toBeInTheDocument();
  });

  it("renders micro-frontends and micro-services tags", () => {
    render(<Experience />);
    expect(screen.getAllByText("Micro-Frontends").length).toBeGreaterThan(0);
    expect(screen.getAllByText("Micro-Services").length).toBeGreaterThan(0);
  });

  it("matches the snapshot structure", () => {
    const { container } = render(<Experience />);
    expect(container).toMatchSnapshot();
  });
});
