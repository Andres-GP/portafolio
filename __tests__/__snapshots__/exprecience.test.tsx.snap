// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Experience /> > matches the snapshot structure 1`] = `
<div>
  <section
    class="py-24"
    id="experience"
  >
    <div
      class="space-y-12"
    >
      <div
        class="space-y-2"
      >
        <h2
          class="text-3xl font-bold tracking-tight text-foreground sm:text-4xl"
        >
          <span
            class="text-primary font-mono text-xl"
          >
            02.
          </span>
           
          Experience
        </h2>
        <div
          class="h-px w-64 bg-border"
        />
      </div>
      <div
        class="space-y-6"
      >
        <div
          data-testid="card"
        >
          <div
            class="space-y-4"
          >
            <div
              class="flex flex-col gap-2 sm:flex-row sm:items-center sm:justify-between"
            >
              <div>
                <h3
                  class="text-xl font-semibold text-foreground"
                >
                  Frontend Developer
                </h3>
                <p
                  class="text-primary"
                >
                  Freelance
                </p>
              </div>
              <p
                class="text-sm text-muted-foreground font-mono"
              >
                March 2025 – Present · Remote
              </p>
            </div>
            <ul
              class="space-y-2"
            >
              <li
                class="flex gap-2 text-sm text-muted-foreground leading-relaxed"
              >
                <span
                  class="text-primary mt-1"
                >
                  ▹
                </span>
                <span>
                  Lead project planning and architecture design for client applications.
                </span>
              </li>
              <li
                class="flex gap-2 text-sm text-muted-foreground leading-relaxed"
              >
                <span
                  class="text-primary mt-1"
                >
                  ▹
                </span>
                <span>
                  Develop and design high-quality web apps with React.js, Next.js, and modern technologies.
                </span>
              </li>
              <li
                class="flex gap-2 text-sm text-muted-foreground leading-relaxed"
              >
                <span
                  class="text-primary mt-1"
                >
                  ▹
                </span>
                <span>
                  Implement scalable micro-frontends and micro-services to ensure flexibility and performance.
                </span>
              </li>
              <li
                class="flex gap-2 text-sm text-muted-foreground leading-relaxed"
              >
                <span
                  class="text-primary mt-1"
                >
                  ▹
                </span>
                <span>
                  Collaborate with clients to deliver intuitive, responsive, and efficient web solutions.
                </span>
              </li>
            </ul>
            <div
              class="flex flex-wrap gap-2"
            >
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                React.js
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Next.js
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                AWS
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Azure
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Node.js
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                TypeScript
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Jest
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Express.js
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Material-UI
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Git
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                GitHub
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                GitLab
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Figma
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                HTML5
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                CSS
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Sass
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Micro-Frontends
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Micro-Services
              </span>
            </div>
          </div>
        </div>
        <div
          data-testid="card"
        >
          <div
            class="space-y-4"
          >
            <div
              class="flex flex-col gap-2 sm:flex-row sm:items-center sm:justify-between"
            >
              <div>
                <h3
                  class="text-xl font-semibold text-foreground"
                >
                  Web UI Developer (Semi-Senior Advanced)
                </h3>
                <p
                  class="text-primary"
                >
                  Globant
                </p>
              </div>
              <p
                class="text-sm text-muted-foreground font-mono"
              >
                Oct 2023 – March 2025 · Remote
              </p>
            </div>
            <ul
              class="space-y-2"
            >
              <li
                class="flex gap-2 text-sm text-muted-foreground leading-relaxed"
              >
                <span
                  class="text-primary mt-1"
                >
                  ▹
                </span>
                <span>
                  Led collaboration with clients through the full development lifecycle — from testing to optimization.
                </span>
              </li>
              <li
                class="flex gap-2 text-sm text-muted-foreground leading-relaxed"
              >
                <span
                  class="text-primary mt-1"
                >
                  ▹
                </span>
                <span>
                  Coached and mentored developers, providing technical feedback and promoting best practices.
                </span>
              </li>
              <li
                class="flex gap-2 text-sm text-muted-foreground leading-relaxed"
              >
                <span
                  class="text-primary mt-1"
                >
                  ▹
                </span>
                <span>
                  Participated in agile project planning and delivery processes to meet product goals.
                </span>
              </li>
              <li
                class="flex gap-2 text-sm text-muted-foreground leading-relaxed"
              >
                <span
                  class="text-primary mt-1"
                >
                  ▹
                </span>
                <span>
                  Drove UI performance improvements and enhanced accessibility in production applications.
                </span>
              </li>
            </ul>
            <div
              class="flex flex-wrap gap-2"
            >
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                React.js
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Next.js
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                TypeScript
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Node.js
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Jest
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Material-UI
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Azure
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                AWS
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                GitLab
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Agile
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                HTML5
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                CSS
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Sass
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Micro-Frontends
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Micro-Services
              </span>
            </div>
          </div>
        </div>
        <div
          data-testid="card"
        >
          <div
            class="space-y-4"
          >
            <div
              class="flex flex-col gap-2 sm:flex-row sm:items-center sm:justify-between"
            >
              <div>
                <h3
                  class="text-xl font-semibold text-foreground"
                >
                  Front-End Developer & UX/UI Designer
                </h3>
                <p
                  class="text-primary"
                >
                  Kovah
                </p>
              </div>
              <p
                class="text-sm text-muted-foreground font-mono"
              >
                Dec 2021 – Oct 2023 · Remote
              </p>
            </div>
            <ul
              class="space-y-2"
            >
              <li
                class="flex gap-2 text-sm text-muted-foreground leading-relaxed"
              >
                <span
                  class="text-primary mt-1"
                >
                  ▹
                </span>
                <span>
                  Developed and optimized large-scale web applications using React with Redux, Sagas, and Tailwind.
                </span>
              </li>
              <li
                class="flex gap-2 text-sm text-muted-foreground leading-relaxed"
              >
                <span
                  class="text-primary mt-1"
                >
                  ▹
                </span>
                <span>
                  Supervised UX/UI design processes to improve usability and visual consistency using Figma.
                </span>
              </li>
              <li
                class="flex gap-2 text-sm text-muted-foreground leading-relaxed"
              >
                <span
                  class="text-primary mt-1"
                >
                  ▹
                </span>
                <span>
                  Coordinated with team members and project managers for continuous delivery and feature planning.
                </span>
              </li>
              <li
                class="flex gap-2 text-sm text-muted-foreground leading-relaxed"
              >
                <span
                  class="text-primary mt-1"
                >
                  ▹
                </span>
                <span>
                  Led technical interviews to expand and strengthen the development team.
                </span>
              </li>
            </ul>
            <div
              class="flex flex-wrap gap-2"
            >
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                React.js
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Redux
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Sass
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Tailwind
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Material-UI
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Semantic UI
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Formik
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Yup
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Emotion
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Framer
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Rollup.js
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Webpack
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                HTML5
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                CSS
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                TypeScript
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                JavaScript
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Micro-Frontends
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Micro-Services
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Figma
              </span>
            </div>
          </div>
        </div>
        <div
          data-testid="card"
        >
          <div
            class="space-y-4"
          >
            <div
              class="flex flex-col gap-2 sm:flex-row sm:items-center sm:justify-between"
            >
              <div>
                <h3
                  class="text-xl font-semibold text-foreground"
                >
                  Front-End Developer
                </h3>
                <p
                  class="text-primary"
                >
                  A&L Software
                </p>
              </div>
              <p
                class="text-sm text-muted-foreground font-mono"
              >
                Jan 2020 – Dec 2021 · Remote
              </p>
            </div>
            <ul
              class="space-y-2"
            >
              <li
                class="flex gap-2 text-sm text-muted-foreground leading-relaxed"
              >
                <span
                  class="text-primary mt-1"
                >
                  ▹
                </span>
                <span>
                  Developed, tested, and optimized React-based applications using modern front-end frameworks.
                </span>
              </li>
              <li
                class="flex gap-2 text-sm text-muted-foreground leading-relaxed"
              >
                <span
                  class="text-primary mt-1"
                >
                  ▹
                </span>
                <span>
                  Collaborated with project managers, stakeholders, and clients to deliver high-quality solutions.
                </span>
              </li>
              <li
                class="flex gap-2 text-sm text-muted-foreground leading-relaxed"
              >
                <span
                  class="text-primary mt-1"
                >
                  ▹
                </span>
                <span>
                  Implemented responsive UIs and ensured cross-browser compatibility using modern web standards.
                </span>
              </li>
              <li
                class="flex gap-2 text-sm text-muted-foreground leading-relaxed"
              >
                <span
                  class="text-primary mt-1"
                >
                  ▹
                </span>
                <span>
                  Worked with Next.js, Tailwind, and Ant Design to enhance performance and maintainability.
                </span>
              </li>
            </ul>
            <div
              class="flex flex-wrap gap-2"
            >
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                React.js
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Next.js
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                TypeScript
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                JavaScript
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Axios
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Ant Design
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Tailwind
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Bootstrap
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Formik
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                Webpack
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                HTML5
              </span>
              <span
                class="rounded-full bg-primary/10 px-3 py-1 text-xs font-mono text-primary"
              >
                CSS
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
